### DISCUSSION TITLE
How to get absolute mouse position when pressing or release mouse buttons?

### DISCUSSION THREAD

**Discussion by shi-yan** [rust-windowing/winit#2280] [Q&A]
The winit api seems to be different from other event apis.

For example, the mouse release and press events only say about the triggering mouse button. but it doesn't allow querying for the mouse cursor's absolute position with respect to the containing window or to the desktop.

The mouse move event, on the other hand, only returns the delta.

What's the recommended way of obtaining the absolution mouse position?

Furthermore, it seems that on wayland, winit doesn't support inner_position and outer_position to get the location of a window.  I'm curious why?

**Comment by madsmtm** [comment]
If you want absolute mouse position within a window, you can use [`WindowEvent::CursorMoved`](https://docs.rs/winit/latest/winit/event/enum.WindowEvent.html#variant.CursorMoved) - true, that doesn't allow querying, so you won't know where the user's cursor is before they start moving it, but that's probably livable?

Absolute mouse position within the desktop can be found by adding [`Window::inner_position`](https://docs.rs/winit/latest/winit/window/struct.Window.html#method.inner_position) to it.
